### ISSUE TITLE
Dynamic requires and bundling

### ISSUE THREAD

**Issue by jeremejevs** [rogerc/file-stream-rotator#35]
Please, use `readFileSync` and `JSON.parse` instead of `require` for audit files here:

https://github.com/rogerc/file-stream-rotator/blob/a968569ee44e21b9ce8ae48e383b8339f41133ca/FileStreamRotator.js#L198-L204

This causes issues with bundlers like Webpack, since they try to be smart about non-constant `require` statements, and start complaining when they can't narrow them down to a reasonable degree. To suppress the complaints, I currently have to use the following Webpack plugin:

```js
new webpack.ContextReplacementPlugin(/file-stream-rotator$/, ctx => {
  for (const x of ctx.dependencies) {
    x.critical = false;
  }
})
```

This wouldn't be necessary with `readFileSync` (which the current usage of `require` is a shortcut for).

**Comment by olsonpm** [comment]
Sounds like a reasonable request.  You should create a PR

**Comment by rogerc** [comment]
I've looked into this. Its an easy fix. Will do a new version with this fix and correct string count for double byte characters over the weekend. thanks.

**Comment by rogerc** [comment]
Published support for Webpack in v0.4.0. Thanks to @DrPlop for the PR.
